modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordApi.kt

```diff
@@ -254,20 +213,21 @@ class MirrordApi(private val service: MirrordProjectService) {
             return
         }
 
-        service.notifier.notification(
-            "Enjoying mirrord? Don't forget to leave a review! Also consider giving us some feedback, we'd highly appreciate it!",
-            NotificationType.INFORMATION
-        )
-            .withLink(
-                "Review",
-                "https://plugins.jetbrains.com/plugin/19772-mirrord/reviews"
-            )
-            .withLink("Feedback", FEEDBACK_URL)
-            .withDontShowAgain(MirrordSettingsState.NotificationId.PLUGIN_REVIEW)
-            .fire()
+        service.notifier.notification("Enjoying mirrord? Don't forget to leave a review! Also consider giving us some feedback, we'd highly appreciate it!", NotificationType.INFORMATION).withLink("Review", "https://plugins.jetbrains.com/plugin/19772-mirrord/reviews").withLink("Feedback", FEEDBACK_URL).withDontShowAgain(MirrordSettingsState.NotificationId.PLUGIN_REVIEW).fire()
     }
 }
 
+/**
+ * Gets the env vars set by the user for the current run (there might be more than 1 run configuration).
+ *
+ * @param project: Contains the `selectedConfiguration`, which holds the active env vars.
+ * @return A `Map` with the launch env vars, that can be put into the `environment` when running mirrord.
+ */
+fun getEnvVarsFromActiveLaunchSettings(project: Project): Map<String, String>? {
```

Comment:
A bit worried about race conditions. Can the user change selected configuration between our entrypoint and this function? There is a `mirrord ls` invocation in between

Reply:
Yeah, but there's `mirrord ls` in between. Depending on cluster connection, it could take seconds. We already pass a part of run config environment (`MIRRORD_CONFIG_FILE`) to our main method, so maybe just replace it with the whole dict?

Reply:
Alright, I'm too dumb to figure this out. Do you mean to add the launch env vars to the various `override fun patchCommandLine` for each IDE type?

I can't see the `MIRRORD_CONFIG_FILE` env var being used anywhere else (and trying to find it in this IDE has not been a fun experience).

Please eli5 and hold my hand, I'm very lost in this kotlin world.

Reply:
It's ok, I as well have fought the IntelliJ war :V

If you look at [`MirrordExecManager.Wrapper.configFromEnv`](https://github.com/metalbear-co/mirrord-intellij/blob/1652125fb763e0b845f2546a14362a5959fcb19e/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt#L162) and its usages, you'll see that we already use some environment from user's run configurations (namely - `MIRRORD_CONFIG_FILE`, which name is defined as const [here](https://github.com/metalbear-co/mirrord-intellij/blob/1652125fb763e0b845f2546a14362a5959fcb19e/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordConfigAPI.kt#L11)). My proposal is to replace this field with `envFromRunConfig: Map<String, String>`, so that you can use the whole env in [`MirrordExecManager.start`](https://github.com/metalbear-co/mirrord-intellij/blob/1652125fb763e0b845f2546a14362a5959fcb19e/modules/core/src/main/kotlin/com/metalbear/mirrord/MirrordExecManager.kt#L93). From there, I'd just pass it down wherever needed.

Reply:
Wow, just wow. Had to make changes in a bunch of places?

I think it's possible to just flat out remove the env vars handling from the project specific classes, as they seemed to be just ways of getting the config file path from the `launch.json` (or whatever name is the equivalent in intellij). But I'm afraid a few do need special treatment?

The env vars mixing gets very confusing real quick, I'm holding myself back from refactoring the whole thing to being loaded once, and we just access it, as in some places we have explicit env var lookups like `env["MIRRORD_DEBUGGER_PORT"]`, which will just not work if the user sets it in `launch.json`.